cpp/daal/include/data_management/data/internal/numeric_table_sycl_homogen.h

```diff
@@ -571,9 +571,49 @@ class DAAL_EXPORT SyclHomogenNumericTable : public SyclNumericTable
 };
 /** @} */
 
+/**
+ * Converts numeric table with arbitrary storage layout to SYCL homogen numeric table of the given type
+ * \param[in]  src               Pointer to numeric table
+ * \return                       Pointer to homogen numeric table
+ */
+template <typename T>
+inline daal::data_management::NumericTablePtr convertToSyclHomogen(NumericTable & src, services::Status & st)
+{
+    using namespace daal::services;
+
+    size_t ncols = src.getNumberOfColumns();
+    size_t nrows = src.getNumberOfRows();
+
+    NumericTablePtr dst = SyclHomogenNumericTable<T>::create(ncols, nrows, NumericTableIface::doAllocate, &st);
+    if (!st.ok())
+    {
+        return daal::data_management::NumericTablePtr();
+    }
```

Comment:
Don't we have macro for that? Consider `DAAL_CHECK_STATUS_RETURN_IF_FAIL`

Reply:
Done